docs/docs/contribute/software/add-new-metric-provider.md

```diff
@@ -0,0 +1,123 @@
+   [metrics-operator/controllers/common/providers](<https://github.com/keptn/lifecycle-toolkit/tree/main/metrics-operator/controllers/common/providers>),
+   matching the new provider name (`placeholder` in our example).
+   Create a new Go package for the placeholder provider in that folder.
+   This package should contain a `struct` that implements the `KeptnSLIProvider` interface.
+   To fully implement the `KeptnSLIProvider` interface, it's necessary to implement the following functions.
+  
+      * `EvaluateQuery`(Fetches metric values from the provider)
+        * This function fetches metric values based on the provided
+          metric query from the provider.
+          It evaluates the query and returns the metric values
+          along with any additional data if required.
+        * It takes as input a [KeptnMetric](../../reference/crd-reference/metric.md)
+          and [KeptnMetricsProvider](../../reference/crd-reference/metricsprovider.md)
+      * `EvaluateQueryForStep`(Fetches metric values with step interval from the provider)
+        * This function fetches metric values with a specified step interval from the placeholder provider.
+          It takes into account the metric query and the step interval provided, executes the query,
+          and returns the metric values along with any additional data if required.
+        * It takes as input a [KeptnMetric](../../reference/crd-reference/metric.md)
+          and [KeptnMetricsProvider](../../reference/crd-reference/metricsprovider.md)
+      * `FetchAnalysisValue`(Fetches analysis values from the provider) functions.
+        * This function fetches analysis values based on the provided query and time range from the
+          provider.
+          It evaluates the query within the specified time range and returns the analysis
+          values along with any additional data if required.
+        * It takes as input an [Analysis](../../reference/crd-reference/analysis.md),
+          resource that contains a `query` and a
+          [KeptnMetricsProvider](../../reference/crd-reference/metricsprovider.md) resource.
+
+   You can follow other existing implementations,
+   such as [prometheus.go](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/controllers/common/providers/prometheus/prometheus.go),
+   as an example.
+
+   **NB:** Each of the three functions expects a string containing a float value in it.
+   But for example purposes we returned some of the data accessible in the function.
+   Below is an example of a placeholder provider implementation.
+
+    ```go
+       {% include "./assets/example-code/placeholder-code-example.go" %}
+    ```
+
+   > **Note** Refer to the documentation of the
+   > [KeptnMetric](https://github.com/keptn/lifecycle-toolkit/blob/main/docs/docs/reference/crd-reference/metric.md)
+   > and
+   > [Analysis](https://github.com/keptn/lifecycle-toolkit/blob/main/docs/docs/reference/crd-reference/analysis.md)
```

Comment:
These should be relative links to these pages.  Check out some of the other docs if you aren't sure how to xref other pages in the docs.

Reply:
I thought i already did that.
I will check it again.